gptlatextranslator/GPTTranslator.py

```diff
@@ -3,21 +3,65 @@
 import warnings
 from typing import List
 
+# For more info on available models, see https://platform.openai.com/docs/models/overview
+MODELS = [
+    "gpt-3.5-turbo",
+    "gpt-3.5-turbo-16k",
+    "gpt-4",
+    "gpt-4-32k"
+  ]
+
+def get_default_max_input_tokens(model_name: str) -> int:
+    """Returns the default maximum number of tokens that can be sent to the model, before running intro troubles."""
+    max_tokens = None
+
+    if model_name == "gpt-3.5-turbo":
+        max_tokens = 4096
+    elif model_name == "gpt-3.5-turbo-16k":
+        max_tokens = 16384
+    elif model_name == "gpt-4":
+        max_tokens = 8192
+    elif model_name == "gpt-4-32k":
+        max_tokens = 32768
+    else:
+        raise ValueError(f"Unknown model name '{model_name}'.")
+
+    # 0.31 is a magic number that allows us enough space to add the system context and the translation prompt, while also taking into account the output of the model.
+    max_input_tokens = int(max_tokens * 0.31)
+    return max_input_tokens
+
 class GPTTranslator:
     """A class for translating LaTex text using OpenAI's API."""
 
     def __init__(self, lang_from:str = "English", 
                  lang_to:str = "Spanish", 
-                 model_name: str = "gpt-3.5-turbo", 
-                 max_input_tokens: int = 1300,
+                 model_name: str = "gpt-3.5-turbo-16k",
                  ignore_commented_blocks: bool = True,
                  verbose: bool = False,
+                 max_input_tokens: int = None,
                  openai_api_key: str = None):
-        
+
         self.model_name = model_name
         self.system_context = f"You are a helpful assistant that translates {lang_from} to {lang_to}."
-        self.translation_prompt = f"Translate the following LaTex text from {lang_from} to {lang_to}:"
-        self.max_text_tokens = max_input_tokens - self._count_tokens(self.translation_prompt) - self._count_tokens(self.system_context)
+        self.translation_prompt = f"Translate the following LaTex code from {lang_from} to {lang_to}. "
```

Comment:
Have you actually tried a translation using this new context and prompt? I worry that given that the prompt says "LaTex code" instead of "LaTex text" the model would also try to translate the LaTex commands. Any thoughts? 

Prompt engineering! wooo!

Reply:
I think we may have that problem in both versions:
- https://chat.openai.com/share/88280b36-4942-4d5f-88de-d74fb058054b
- https://chat.openai.com/share/ae752a56-932a-4d10-a2ee-e01b37252dc0

In any case, I'll rollback that change, since I liked it better with "text" instead of "code".